### ISSUE TITLE
Report nullability warning for `foreach` of nullable value type

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#31503]
Should report a warning for the implicit call to `s.Value.GetEnumerator()`:
```C#
using System.Collections;

struct S : IEnumerable
{
    public IEnumerator GetEnumerator() => throw null;
}

class Program
{
    static void F(S? s)
    {
        foreach (var i in s)
        {
        }
    }

    static void Main()
    {
        F(null);
    }
}
```

**Comment by cston** [comment]
@alr, thanks for reporting that issue. It looks like it has been fixed in #31171.

**Comment by alrz** [comment]
@cston besides of that I didn't see any test that would warn a possible null dereference on `await x?.M()`

**Comment by jcouv** [comment]
I verified that the `foreach` scenario is covered (see `ForEach_12` ... `ForEach_15`).
The corresponding issue with `await` was fixed in https://github.com/dotnet/roslyn/pull/31981

Closing the issue as resolved. Thanks

**Comment by cston** [comment]
The original issue with `foreach` still needs to be add. See `NullableReferenceTypesTests.NullableT_ForEach`.
